Add community tutorials, disable warnings in conditional compilation

danwilhelm opened the pull request:
- Add community tutorials section to guide.
- Disable conditional compilation warning (to resolve unused variable warning on non-Windows from #773).
- Update changelog.

mitchmindtree commented on guide/src/welcome.md:
```diff
@@ -31,6 +31,8 @@ A suite of tutorials demonstrating how to do different things with Nannou. For
 example, "How do I output sounds?", "How do I draw shapes?", "How can I connect
 to my laser?"
 
+	- [Community Tutorials](/community_tutorials.md)
```
This will end up formatted a little strangely - maybe instead we can add another sentence along the lines of

> You can find more tutorials by the community [here](./community_tutorials.md).

mitchmindtree commented on nannou/src/draw/renderer/mod.rs:
```diff
@@ -974,6 +974,18 @@ impl Renderer {
+                    state.background_color = Some(Default::default());
+                }
+            }
+        }
```
This might sound a little crazy, but I think we should also consider the case where a user might have a transparent window - is it still possible with a setup like this? I guess it might work still as long as `background_color` is set to a zero-alpha value?

danwilhelm replied:
I verified that if the window is transparent, clearing it with alpha=1. results in a colored background, while alpha=0. results in transparent. So I'll move this code into `to_frame()` and set the alpha based on whether the window is transparent.

danwilhelm replied:
If you resize the window or restore it later, I'm unsure whether the frame needs to be re-cleared. I noticed the frame &  window counters just continually increase no matter what, so they wouldn't be useful in that situation.

mitchmindtree replied:
> If you resize the window or restore it later, I'm unsure whether the frame needs to be re-cleared

Yeah I think clearing on the first frame is probably enough, it's not super clear that there's a single correct behaviour for most users when it comes to resizing (e.g. we might clear over a long-running additive sketch), so it might be best to leave that case up to the user.

danwilhelm replied:
I noticed it always loses the old frame data upon resize, because it always allocates a new frame buffer. So I concluded it does need clearing on the first frame of each new framebuffer.

This seems consistent with the recent user-reported issue on Matrix, who said the fullscreen transition happened on about the tenth frame. It only showed uninitialized contents (weird color tint) when it moved to fullscreen.

mitchmindtree reviewed (commented):
This is a great idea! Nice one @danwilhelm :)

danwilhelm commented:
I believe there is something wrong with the automated checks -- cannot compile `hotglsl`?

- Moved the background clear into another PR, since I realized there are additional cases to handle. 
- Updated the community tutorial reference as suggested to a `>` rather than a `-`.

mitchmindtree commented:
> I believe there is something wrong with the automated checks -- cannot compile hotglsl?

Ahhh looks like there might have been a breaking change upstream in `notify`, I'll do some digging. I think this is fine to merge :)
